w_common/lib/invalidation_mixin.dart

```diff
@@ -35,19 +35,19 @@ abstract class InvalidationMixin {
   /// and [ValidationStatus.cancelled] if invalidation is cancelled.
   Future<ValidationStatus> invalidate() {
     if (invalid) {
-      return _onValidate.future;
+      return _onValidate!.future;
     }
 
     _onValidate = Completer<ValidationStatus>();
 
     window.animationFrame.then((_) {
       if (invalid) {
         validate();
-        _onValidate.complete(ValidationStatus.complete);
+        _onValidate!.complete(ValidationStatus.complete);
```

Comment:
I think this instance should use ? rather than !. Another way to approach this code is to keep a local `final onValidate` that we use here.